project/buildSrc/src/main/kotlin/db.kt

```diff
@@ -265,103 +269,107 @@ open class PgAddData : AbstractTask() {
     fun run() {
-       Token().create("urn:org:ionproject:scopes:api:read", project, false)
+        Token().create("urn:org:ionproject:scopes:api:write")
     }
 }
 
-open class PgInsertIssueTokenWin : AbstractTask() {
+open class PgInsertIssueToken : AbstractTask() {
     @TaskAction
     fun run() {
-        Token().create("urn:org:ionproject:scopes:token:issue", project, false)
+        Token().create("urn:org:ionproject:scopes:token:issue")
     }
 }
 
+private data class TokenDbParams(
+  val hash: String,
+  val isValid: Boolean,
+  val issuedAt: Long,
+  val expiredAt: Long,
+  val scope: String,
+  val clientId: Int,
+  val base64Token: String)
 
 class Token {
-    fun create(scope: String, project: Project, isPosixShell: Boolean) {
-        val values = if (isPosixShell) getTokenReferences(scope) else getTokenReferencesWin(scope)
-
-        val insertQuery = values[0]
-        val base64Reference = values[1]
-
-        val pgParams = Postgres.pgParams
-        val result = project.exec {
-            commandLine("docker", "run",
-                "-e", "PGPASSWORD=${pgParams.password}",
-                "--rm", "--network=host", Docker.IMAGE_NAME,
-                "psql",
-                "-h", pgParams.host,
-                "-U", pgParams.user,
-                "-d", pgParams.db,
-                "-p", pgParams.port,
-                "-w",
-                "-1",
-                "-c",
-                insertQuery)
-
-            environment(Postgres.ENV_PASSWORD, pgParams.password)
-        }
-        result.assertNormalExitValue()
-        print("Your token reference is $base64Reference")
-    }
-
-    private fun getTokenReferences(scope: String): List<String> {
-        val tokenGenerator = TokenGenerator()
-        val randomString = tokenGenerator.generateRandomString()
-        val base64Token = tokenGenerator.encodeBase64url(randomString)
-        val tokenHash = tokenGenerator.getHash(randomString)
-
-        val currTime = System.currentTimeMillis()
-        val expirationTime = currTime + 1000*60*60
-        val claims = """{"scope":"$scope","client_id":500}"""
-
-        val insertQuery = """
-            INSERT INTO dbo.Token(hash,isValid,issuedAt,expiresAt,claims) VALUES ('$tokenHash',true,$currTime,$expirationTime,'$claims')
-        """
-
-        return listOf(insertQuery, base64Token)
+    fun create(scope: String) {
+        val params = getTokenReferences(scope)
+        val cid = 500
+        val json = PGobject(); json.type = "jsonb"; json.value = """{"scope":"$scope", "client_id": ${cid}}"""
+        val con = Db.getConnection()
+        val st = con.prepareStatement("INSERT INTO dbo.Token(hash,isValid,issuedAt,expiresAt,claims) VALUES (?,?,?,?,?);")
+        st.setString(1, params.hash)
+        st.setBoolean(2, params.isValid)
+        st.setLong(3, params.issuedAt)
+        st.setLong(4, params.expiredAt)
+        st.setObject(5, json)
+        print("Your token reference is ${params.base64Token}")
```

Comment:
The SQL statement is not being executed and the connection is not being closed.

Reply:
Yickes. Completely forgot. It should execute though?

Reply:
🤔 If we want to do something I think it should.

Reply:
Haha. Turns out, I removed the line that executed the statement just as I committed the changes. *brain dead*